File: BotsSDK/korebotsdklib/src/main/java/kore/botssdk/models/BrandingOverrideConfigModel.java
package kore.botssdk.models;

public class BrandingOverrideConfigModel {
    private Boolean enable;
    private Boolean emoji_short_cut;
    private int typing_indicator_timeout;
    private BrandingOverrideHistoryModel history;

    public BrandingOverrideConfigModel updateWith(BrandingOverrideConfigModel configModel) {
        enable = configModel.enable != null ? configModel.enable : enable;
        emoji_short_cut = configModel.emoji_short_cut != null ? configModel.emoji_short_cut : emoji_short_cut;
        history = history != null && configModel.history != null ? history.updateWith(configModel.history) : history;
        return this;
    }

    public BrandingOverrideHistoryModel getHistory() {
        return history;
    }

    public int getTyping_indicator_timeout() {
        return typing_indicator_timeout;
    }

    public boolean isEnable() {
        return enable != null ? enable : false;
    }

    public boolean isEmoji_short_cut() {
        return emoji_short_cut != null ? emoji_short_cut : false;
    }
}

File: BotSdk/botclient/src/main/java/com/kore/network/api/responsemodels/branding/BrandingOverrideConfigModel.kt
package com.kore.network.api.responsemodels.branding

import com.google.gson.annotations.SerializedName

data class BrandingOverrideConfigModel(
    @SerializedName("isEnable")
    val isEnable: Boolean? = false,
    @SerializedName("isEmoji_short_cut")
    val emojiShortCut: Boolean? = false,
    @SerializedName("typing_indicator_timeout")
    val typingIndicatorTimeout: Int? = 0,
    @SerializedName("history")
    val history: BrandingOverrideHistoryModel? = null
)

fun BrandingOverrideConfigModel.updateWith(configModel: BrandingOverrideConfigModel?): BrandingOverrideConfigModel {
    return this.copy(
        isEnable = configModel?.isEnable ?: isEnable,
        emojiShortCut = configModel?.emojiShortCut ?: emojiShortCut,
        typingIndicatorTimeout = configModel?.typingIndicatorTimeout ?: typingIndicatorTimeout,
        history = history?.updateWith(configModel?.history) ?: history
    )
}
